interfaces/cython/cantera/ck2cti.py

```diff
@@ -312,14 +316,55 @@ def to_cti(self, indent=0):
         else:
             optStr = repr(options)
 
-        if self.duplicate:
-            kinstr = kinstr[:-1] + ",\n{0}options={1})".format(k_indent, optStr)
-
         if self.fwdOrders:
             order = ' '.join('{0}:{1}'.format(k,v)
                              for (k,v) in self.fwdOrders.items())
-            kinstr = kinstr[:-1] + ",\n{0}order='{1}')".format(k_indent, order)
 
+            for(k,v) in self.fwdOrders.items():
+                #Check for negative orders
+                if float(v) < 0 and NEGATIVE_ORDERS is False:
+                    #Meaning negative order detected but --allow-negative-order flag not passed to ck2cti
+                    print('\nWARNING: Reaction with negative order detected.') 
+                    print('Re-execute ck2cti with --allow-negative-orders and also execute ck2cti --help to see note on danger of negative orders.\n')
+                elif float(v) < 0 and NEGATIVE_ORDERS:
+                    #Meaning negative order detected and user is aware
+                    options.append('negative_orders')
+                    if len(options) == 1:
+                        optStr = repr(options[0])
+                    else:
+                        optStr = repr(options)
+                    break
+                
+            for(k,v) in self.fwdOrders.items():
+                #Check for non-reactant orders
+                if k not in self.reactantString and NON_REACTANT_ORDERS is True:
+                    options.append('nonreactant_orders')
+                    if len(options) == 1:
+                        optStr = repr(options[0])
+                    else:
+                        optStr = repr(options)
+                    break
+                elif k not in self.reactantString and NON_REACTANT_ORDERS is False:
+                    print('\nWARNING: Reaction detected with non-reactant order.')
+                    print('Re-execute ck2cti with --allow-nonreactant-orders and also execute ck2cti --help to see note on danger of non-reactant orders.\n')
+                else:
+                    pass
+                
+            if 'negative_orders' in options:
+                kinstr = kinstr[:-1] + ",\n{0}order='{1}', options={2})".format(k_indent, order, optStr)
+            else:
+                kinstr = kinstr[:-1] + ",\n{0}order='{1}')".format(k_indent, order)
+
+        if self.duplicate and 'negative_orders' not in options:
+            kinstr = kinstr[:-1] + ",\n{0}options={1})".format(k_indent, optStr)
+            
+                
+        if len(options) == 1:
```

Comment:
I think the logic for creating `optStr` needs to be needs to be re-examined. In the loops above, you can simply update `options`, then `optStr` can be generated just once. Then, the condition for appending an `options=...` section to `kinStr` should be conditioned simply on `options` being non-empty (i.e. just `if options`: ...).